use auto & cell instead of auto cell

tamiko opened the pull request:
Unify the range-based for loops over active cells as discussed in #8102

tamiko commented:
/rebuild

kronbichler commented on examples/step-49/doc/results.dox:
```diff
@@ -271,7 +271,7 @@ Triangulation<3> triangulation;
 triangulation.set_manifold(cylinder_id, cylinder);
 
-for (auto cell : triangulation.active_cell_iterators())
+for (auto & cell : triangulation.active_cell_iterators())
```
Is there a reason why clang-format sometimes puts a space between `& cell` and sometimes not?

bangerth replied:
This is in a .dox file. clang-format doesn't touch those. I suspect that @tamiko used the version with the space in his regex-replace, and then clang-format removed the space in all places it gets to see.

tamiko replied:
Exactly

luca-heltai commented:
Wouldn't `const auto &cell` be even better?

bangerth commented on include/deal.II/differentiation/ad/ad_helpers.h:
```diff
@@ -1081,7 +1081,7 @@ namespace Differentiation
      *
      *   // Assembly loop:
-     *   for (auto cell & : ...)
+     *   for (auto & cell : ...)
```
ha!

jppelteret replied:
whoops!

bangerth commented:
On 5/12/19 2:27 AM, Luca Heltai wrote:
> Wouldn't |const auto &cell| be even better?

That's what I have been writing for the most part.

tamiko commented:
I have updated this PR (fixing the code snippets in the dox files). I have removed the change in `dof_handler_policy.cc` which will hopefully make the CI happy.
